internal/storage/gcs/bucket.go

```diff
@@ -44,6 +45,9 @@ type Bucket interface {
 	// Return Type of bucket e.g. Hierarchical or NonHierarchical
 	BucketType() BucketType
 
+	// Fetch bucket type from GetStorageLayout and set it. e.g. Hierarchical or NonHierarchical
```

Comment:
Is it necessary to add a new method in the bucket interface?

Can't we set at the first BucketType() method call, or during initialization of bucket handle?

Reply:
All the clients and control client method are in bucket_handle and get called by  bucket_handle interface only. So to keep consistency we decided that.

Reply:
can we call FetchAndSetBucketType [here](https://github.com/GoogleCloudPlatform/gcsfuse/blob/f86b35dd7d21b90d0b41fda56e736a310304e881/internal/gcsx/bucket_manager.go#L167), before b is converted to corresponding bucket, that way only bucket handle needs to have method, thoughts ? @Tulsishah @raj-prince

Reply:
No we can't do that because b type is gcs.Bucket. So it have to part of bucket interface to call it from setUpBucket.

Reply:
got it, lets keep it in all bucket types then, regarding two alternatives:
set at first buckettype --> this will slow down the first corresponding operation which will trigger bucket type, which is not expected
initialization of bucket handle--> bucket handle is being initialised in storage handle, storage handle is not the correct domain to call storage layout api, setup bucket method should only know how to fetch type and set for the bucket, this behaviour should be abstracted from storage handl, keeping it in storage handle breaks abstraction principle.

Reply:
Had offline discussion to merge method in BucketType() only.